frontend/src/metabase/dashboard/components/add-card-sidebar/QuestionList.tsx

```diff
@@ -45,16 +60,25 @@ export function QuestionList({
   const trimmedSearchText = searchText.trim();
   const isSearching = !!trimmedSearchText;
 
-  let query = isSearching
-    ? { q: trimmedSearchText }
-    : { collection: collectionId };
+  const query = createQuery();
 
-  query = {
-    ...query,
-    models: ["card", "dataset"],
-    offset: queryOffset,
-    limit: DEFAULT_SEARCH_LIMIT,
-  };
+  function createQuery(): SearchListQuery {
```

Comment:
I was on the fence about that since it will need a lot of parameters.

```ts
function createQuery(
  isSearching: boolean,
  trimmedSearchText: string,
  showOnlyPublicCollections: boolean,
  collectionId: string | number,
  queryOffset: number,
) {
```

it feels like with this many parameters, it'd be easier to just inline it (or have a function inside the component as I did)